nodejs/week2/src/backend/routes/reviews.js

```diff
@@ -0,0 +1,36 @@
+const express = require('express');
+const router = express.Router()
+const ReviewsArray = require('../data/reviews.json');
+
+
+
+router.get('/reviews', (req, res)=>{
+        
+// ===================================================================================  for id :  /reviews?id=2
+      const givenid = req.query.id;
+      
+        let reviewscount = Object.keys(ReviewsArray).length;
+        const calledreviews = ReviewsArray.filter(item => {
+                if(reviewscount < givenid){
+                    return ("This review does not exist");

```

Comment:
This will be interpreted as `true`. `filter` function has to always return boolean.

Reply:
Sorry, but here it says in technical details that it returns an array containing all the array elements that pass the test. If no elements pass the test it returns an empty array.
https://www.w3schools.com/jsref/jsref_filter.asp